sheepdog/transactions/upload/sub_entities.py

```diff
@@ -711,7 +713,18 @@ def _version_index(self, **kwargs):
 
     def _should_version_node(self, node):
         """ Performs checks to determine if the current entity should be versioned"""
-        return self._is_modified_release_node(node)
+        return self._is_new_release(node) and self._is_modified_release_node(node)
+
+    def _is_new_release(self, node):
+        """ Checks if this node should be treated as part of a new release. If the current release number on this node
+            is not less than the active release number, then it is not a new release
+         """
+        doc = self.get_indexed_document()
+        if doc and node and node.state in RELEASED_NODE_STATES:
+            current_release = doc.metadata.get("release_number", None)
+            return current_release is not None and (self.transaction.active_data_release is None or StrictVersion(
+                self.transaction.active_data_release) > StrictVersion(current_release))
```

Comment:
It looks like the order of operations is correct here, but can you put an extra set of parens around the StrictVersion greater than comparison to make it clearer? That line is pretty dense already.